UI/src/utils/styleRule.js

```diff
@@ -68,7 +68,16 @@
       if(!expression || expression.compare)
         throw 'You must have on of the following functions ["where", "and", "or"] before use the "is".';
       expression.compare = function(propertyValue){
-        return propertyValue == value;
+        // Can be improved - solution found for obstacles since they are the only ones that have more than one two colors on point assets
+        var property;
+
+        if(!_.isUndefined(propertyValue) && propertyValue.constructor === Array) {
+          property = _.head(_.find(propertyValue, function(property){ return property.publicId === "esterakennelma"; }).values).propertyValue;
+        }
+
+        var finalProperty = property ? property : propertyValue;
+
+        return finalProperty == value;
```

Comment:
At least we could extend styleRule.js and create a separate one for obstacles where the 'is' function is always done like this.